protocol/x/ratelimit/keeper/keeper_test.go

```diff
@@ -51,72 +58,310 @@ func TestSetGetDenomCapacity(t *testing.T) {
-	expectedDenomCapacity := types.DenomCapacity{
-		Denom: testDenom,
-		CapacityList: []dtypes.SerializableInt{
-			// TODO(CORE-834): Update expected value after `GetBaseline` depends on current TVL.
-			dtypes.NewInt(200_000_000_000),   // 200k tokens (assuming 6 decimals)
-			dtypes.NewInt(2_000_000_000_000), // 1m tokens (assuming 6 decimals)
-		},
-	}
-	require.Equal(t, expectedDenomCapacity, gotDenomCapacity, "retrieved DenomCapacity does not match the set value")
+			// Test SetLimitParams
+			k.SetLimitParams(ctx, limitParams)
 
-	// Set empty `LimitParams` for `testDenom`.
-	k.SetLimitParams(ctx, types.LimitParams{
-		Denom:    testDenom,
-		Limiters: []types.Limiter{}, // Empty list, results in deletion of the key.
-	})
+			// Test GetLimitParams
+			gotLimitParams := k.GetLimitParams(ctx, tc.denom)
+			require.Equal(t, limitParams, gotLimitParams, "retrieved LimitParams do not match the set value")
 
-	// Check that the key is deleted under `LimitParams` storage.
-	require.Equal(t,
-		types.LimitParams{
-			Denom:    testDenom,
-			Limiters: nil,
-		},
-		k.GetLimitParams(ctx, testDenom),
-		"retrieved LimitParams do not match the set value")
+			// Query for `DenomCapacity` of `testDenom`.
+			gotDenomCapacity := k.GetDenomCapacity(ctx, tc.denom)
+			// Expected `DenomCapacity` is initialized such that each capacity is equal to the baseline.
+			expectedDenomCapacity := types.DenomCapacity{
+				Denom:        tc.denom,
+				CapacityList: tc.expectedCapacityList,
+			}
+			require.Equal(t, expectedDenomCapacity, gotDenomCapacity, "retrieved DenomCapacity does not match the set value")
 
-	// Check that the key is deleted under `DenomCapacity` storage.
-	require.Equal(t,
-		types.DenomCapacity{
-			Denom:        testDenom,
-			CapacityList: nil,
-		},
-		k.GetDenomCapacity(ctx, testDenom),
-		"retrieved LimitParams do not match the set value")
+			// Set empty `LimitParams` for `testDenom`.
+			k.SetLimitParams(ctx, types.LimitParams{
+				Denom:    tc.denom,
+				Limiters: []types.Limiter{}, // Empty list, results in deletion of the key.
+			})
+
+			// Check that the key is deleted under `LimitParams` storage.
+			require.Equal(t,
+				types.LimitParams{
+					Denom:    tc.denom,
+					Limiters: nil,
+				},
+				k.GetLimitParams(ctx, tc.denom),
+				"retrieved LimitParams do not match the set value")
+
+			// Check that the key is deleted under `DenomCapacity` storage.
+			require.Equal(t,
+				types.DenomCapacity{
+					Denom:        tc.denom,
+					CapacityList: nil,
+				},
+				k.GetDenomCapacity(ctx, tc.denom),
+				"retrieved LimitParams do not match the set value")
+		})
+	}
 }
 
 func TestGetBaseline(t *testing.T) {
-	// TODO(CORE-836): Add test for GetBaseline.
+	tests := map[string]struct {
+		denom            string
+		balances         []banktypes.Balance
+		limiter          types.Limiter
+		expectedBaseline *big.Int
+	}{
+		"max(1% of TVL, 100k token), TVL = 5M token": {
```

Comment:
similar to above, 
1. can you add test cases that do not divide evenly and would cause rounding down and rounding up?
2. as a corner case, can we add test cases that results in 0 baseline?

Reply:
1. Added such cases
2. As mentioned above, we will not allow `0` baseline or `ppm`